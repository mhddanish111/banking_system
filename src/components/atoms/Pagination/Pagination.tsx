import React from 'react';
import { usePagination, DOTS } from '../../../hooks/usePagination';
import './Pagination.scss';

type PaginationProps = {
  totalCount?: number;
  siblingCount?: number;
  currentPage?: number;
  pageSize?: number;
  className?: string;
  onPageChange?: any;
};
const Pagination: React.FC<PaginationProps> = (props: PaginationProps) => {
  const {
    onPageChange,
    totalCount,
    siblingCount = 1,
    currentPage = 0,
    pageSize,
    className,
  } = props;

  const paginationRange =
    usePagination({
      currentPage,
      totalCount,
      siblingCount,
      pageSize,
    }) || [];

  if (currentPage === 0 || paginationRange.length < 2) {
    return null;
  }

  const onNext = (e: any) => {
    e.preventDefault();
    onPageChange(currentPage + 1);
  };

  const onPrevious = (e: any) => {
    e.preventDefault();
    onPageChange(currentPage - 1);
  };

  const getSelected = (pageNumber: any): string => {
    return pageNumber === currentPage ? 'selected' : '';
  };
  const lastPage = paginationRange[paginationRange.length - 1];
  const disabled = currentPage === 1 ? 'disabled' : '';
  const paginationItemDisabled = currentPage === lastPage ? 'disabled' : '';
  return (
    <ul className={`pagination-container ${className}`}>
      <li
        className={`pagination-item ${disabled} arrow-left`}
        onClick={(e) => onPrevious(e)}
        key="arrow-left"
      >
        <div className="arrow left" />
      </li>
      {paginationRange.map((pageNumber) => {
        if (pageNumber === DOTS) {
          return <li className="pagination-item dots">&#8230;</li>;
        }

        return (
          <li
            className={`pagination-item ${getSelected(pageNumber)}`}
            onClick={(e) => {
              e.preventDefault();
              onPageChange(pageNumber);
            }}
            key={`pagination-item-${pageNumber}`}
          >
            {pageNumber}
          </li>
        );
      })}
      <li
        className={`pagination-item ${paginationItemDisabled} arrow-right`}
        onClick={(e) => onNext(e)}
        key="arrow-right"
      >
        <div className="arrow right" />
      </li>
    </ul>
  );
};

export default Pagination;
