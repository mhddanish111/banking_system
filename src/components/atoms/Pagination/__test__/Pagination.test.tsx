import { shallow } from 'enzyme';
import Pagination from '../Pagination';

describe('<Pagination />', () => {
  const props = {
    className: '',
    totalCount: 50,
    currentPage: 1,
    pageSize: 10,
    onPageChange: jest.fn(),
  };
  const wrapper = shallow(<Pagination {...props}>Heading</Pagination>);
  it('should render the Pagination Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call next button', () => {
    const onNext = jest.fn();
    wrapper.find('.arrow-right').simulate('onClick', onNext());
    expect(onNext).toHaveBeenCalled();
  });
  it('should call previous button', () => {
    const onPrevious = jest.fn();
    wrapper.find('.arrow-left').simulate('onClick', onPrevious());
    expect(onPrevious).toHaveBeenCalled();
  });
  it('should call pagination item button', () => {
    const onPageChange = jest.fn();
    wrapper.find('.selected').simulate('onClick', onPageChange(1));
    expect(onPageChange).toHaveBeenCalled();
  });
});
