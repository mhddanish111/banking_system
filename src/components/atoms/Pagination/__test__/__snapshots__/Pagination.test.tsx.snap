// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render the Pagination Component correctly 1`] = `
<ul
  className="pagination-container "
>
  <li
    className="pagination-item disabled arrow-left"
    key="arrow-left"
    onClick={[Function]}
  >
    <div
      className="arrow left"
    />
  </li>
  <li
    className="pagination-item selected"
    key="pagination-item-1"
    onClick={[Function]}
  >
    1
  </li>
  <li
    className="pagination-item "
    key="pagination-item-2"
    onClick={[Function]}
  >
    2
  </li>
  <li
    className="pagination-item "
    key="pagination-item-3"
    onClick={[Function]}
  >
    3
  </li>
  <li
    className="pagination-item "
    key="pagination-item-4"
    onClick={[Function]}
  >
    4
  </li>
  <li
    className="pagination-item "
    key="pagination-item-5"
    onClick={[Function]}
  >
    5
  </li>
  <li
    className="pagination-item  arrow-right"
    key="arrow-right"
    onClick={[Function]}
  >
    <div
      className="arrow right"
    />
  </li>
</ul>
`;
